presto-docs/src/main/sphinx/develop/client-protocol.rst

```diff
@@ -0,0 +1,168 @@
+``X-Presto-Set-Catalog``               Instructs the client to set the catalog that will be sent in the
+                                       ``X-Presto-Catalog`` request header in subsequent client requests.
+``X-Presto-Set-Schema``                Instructs the client to set the schema that will be sent in the ``X-Presto-Schema`` request
+                                       header in subsequent client requests.
+``X-Presto-Set-Session``               The value of the ``X-Presto-Set-Session`` response header is a name=value string,
+                                       representing session attributes that are meaningful to the Presto engine or a connector.
+                                       Instructs the client to add that name=value string to the ``X-Presto-Session``
+                                       request header to be used in subsequent client requests.
+``X-Presto-Clear-Session``             Instructs the client to remove the session property with the whose name is the value
+                                       of the ``X-Presto-Clear-Session`` header from the comma-separated list of session properties
+                                       that will be sent in the ``X-Presto-Session`` header in subsequent client requests.
+``X-Presto-Set-Role``                  Instructs the client to set ``X-Presto-Role`` request header to the catalog role give by the
+                                       value of the ``X-Presto-Set-Role`` headerin subsequent client requests.
+``X-Presto-Added-Prepare``             Instructs the client to add the name=value pair to the set of prepared statements
+                                       that will be sent in the ``X-Presto-Prepared-Statements`` request header
+                                       in subsequent client requests.
+``X-Presto-Deallocated-Prepare``       Instructs the client to remove the prepared statement whose name is the value of the
+                                       ``X-Presto-Deallocated-Prepare`` header from the client's list of prepared statements
+                                       sent in the ``X-Presto-Prepared-Statements`` request header in subsequent client requests.
+``X-Presto-Started-Transaction-Id``    Provides the transaction ID that the client should pass back in the
+                                       ``X-Presto-Transaction-Id`` request header in subsequent requests.
+``X-Presto-Clear-Transaction-Id``      Instructs the client to clear the ``X-Presto-Transaction-Id`` request header used in
+                                       subsequent requests.
+====================================== =================================================================================================
+
+
+Important Presto Client Classes
+===============================
+
+The following Java classes provide exhaustive specifications of the objects exchanged in the Presto client protocol:
+
+* ``QueryResults``
```

Comment:
I would consider linking these to the code in Github.

Reply:
I'd prefer not to have github links that will fail when packages are changed.  Instead I added a table of documentation for the important members of `QueryResults`, and eliminated the class list.